packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -386,21 +386,29 @@ public boolean onRequestPermissionsResult(
       case REQUEST_EXTERNAL_IMAGE_STORAGE_PERMISSION:
         if (permissionGranted) {
           launchPickImageFromGalleryIntent();
+        } else {
+          finishWithError("photo_access_denied", "The user did not allow photo access.");
```

Comment:
maybe combine these 4 statements in the if (!permissionGranted) clause?

Reply:
Refactored per yr comment